source/PlayerInfo.cpp

```diff
@@ -1865,11 +1865,18 @@ const CargoHold &PlayerInfo::DistributeCargo()
 			}
 			else
 			{
-				// Your flagship takes first priority for passengers but last for cargo.
+				// Your flagship takes first priority for passengers but last for cargo from
+				// non-unique missions.
 				desiredCrew = ship->Crew();
 				ship->Cargo().SetBunks(ship->Attributes().Get("bunks") - desiredCrew);
 				for(const auto &it : cargo.PassengerList())
 					cargo.TransferPassengers(it.first, it.second, ship->Cargo());
+				// Load unique mission cargo onto the flagship.
+				for(const auto &it : cargo.MissionCargo())
+				{
+					if(it.first->IsUnique())
```

Comment:
This only checks if the mission is not repeatable. IMO, a new mission attribute, or an attribute of mission cargo would fit better here. Or at least make it a preference, as it was suggested in #6444.

Reply:
I would like all mission cargo to prioritise the flagship. Other than when plundering disabled ships, there is no downside to putting all important stuff on your flagship in preference to others. I would then like it sorted by 'combat survivability' so when my Interceptors die, I only loose some random commodity not mission cargo.

Reply:
> This only checks if the mission is not repeatable. IMO, a new mission attribute, or an attribute of mission cargo would fit better here.

I can also introduce a new attribute, but repeatability is arguably what matters here. If you can repeat a mission, it's OK if you fail it.

> Or at least make it a preference, as it was suggested in #6444.

It could be a preference as long as the default is "load mission cargo onto flagship". Preferences should default to behavior that's safe rather than optimal for specific use cases.

Reply:
> I would like all mission cargo to prioritise the flagship.

I agree that would make sense, but the code history shows that previously, other people strongly wanted the opposite to be the case.